pkg/ccl/streamingccl/streamingest/stream_ingestion_job.go

```diff
@@ -239,6 +230,19 @@ func completeIngestion(
 	}
 	return nil
 }
+
+func completeProducerJob(
+	ctx context.Context, ingestionJob *jobs.Job, internalDB *sql.InternalDB,
+) error {
+	client, err := connectToActiveClient(ctx, ingestionJob, internalDB)
+	if err != nil {
+		return err
+	}
+	streamID := ingestionJob.Details().(jobspb.StreamIngestionDetails).StreamID
+	return errors.CombineErrors(client.Complete(ctx, streampb.StreamID(streamID),
+		true /* successfulIngestion */), client.Close(ctx))
```

Comment:
are you sure you want to return an error when Complete succeeds and Close fails? looks like in the other case we just log.Warning the Close() error.

Reply:
ah, this is sloppy. I pushed the time out into the function and de-duped the OnFailOrCancel method. Now any error during producer job completion gets logged.

Reply:
I see. Your change makes sense to me. I think there is no real need to `CombineErrors` because you don't return a single error, so you can simplify things a tiny bit and just log.Warning the first error and then the second. But that's up to you.

Reply:
yeah, you're right. just changed it.